File: src/main/java/com/profe_me_quiero_graduar/app/service/mapper/ComentariosMapper.java
package com.profe_me_quiero_graduar.app.service.mapper;

import com.profe_me_quiero_graduar.app.domain.*;
import com.profe_me_quiero_graduar.app.service.dto.ComentariosDTO;
import org.mapstruct.*;

/**
 * Mapper for the entity {@link Comentarios} and its DTO {@link ComentariosDTO}.
 */
@Mapper(componentModel = "spring", uses = { AnunciosMapper.class })
public interface ComentariosMapper extends EntityMapper<ComentariosDTO, Comentarios> {
    @Mapping(target = "anuncio", source = "anuncio", qualifiedByName = "id")
    ComentariosDTO toDto(Comentarios s);
}

File: server/src/service/mapper/comentarios.mapper.ts
import { Comentarios } from '../../domain/comentarios.entity';
import { ComentariosDTO } from '../dto/comentarios.dto';

/**
 * A Comentarios mapper object.
 */
export class ComentariosMapper {
    static fromDTOtoEntity(entityDTO: ComentariosDTO): Comentarios {
        if (!entityDTO) {
            return;
        }
        let entity = new Comentarios();
        const fields = Object.getOwnPropertyNames(entityDTO);
        fields.forEach((field) => {
            entity[field] = entityDTO[field];
        });
        return entity;
    }

    static fromEntityToDTO(entity: Comentarios): ComentariosDTO {
        if (!entity) {
            return;
        }
        let entityDTO = new ComentariosDTO();

        const fields = Object.getOwnPropertyNames(entity);

        fields.forEach((field) => {
            entityDTO[field] = entity[field];
        });

        return entityDTO;
    }
}
